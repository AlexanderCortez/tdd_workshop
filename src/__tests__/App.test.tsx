import App from "../App";
import { shallow, render, mount } from "enzyme";

describe("App Test Component", () => {
  it("should render react link", () => {
    const wrapper = shallow(<App />);
    const link = wrapper.find(".App-link");
    expect(link).toHaveLength(1);
    expect(wrapper.text()).toMatch(/learn react/i);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
