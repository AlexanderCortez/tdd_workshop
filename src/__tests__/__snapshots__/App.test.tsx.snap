// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Component should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <span>
    Extra content
  </span>
</div>
`;
